### PR TITLE
Added SELU, ELU and Swish as activation functions for Gluon Interface

### PR DESCRIPTION
**Pull request by anjishnu**
## Description ##
added SELU and ELU activation functions as recommended on this thread https://github.com/apache/incubator-mxnet/issues/8422

### Essentials ###
- [x]  Passed code style checking (`make lint`)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by dongzhuoyao** [comment]
well done!

**Comment by kobenaxie** [comment]
Why not use 'HybridBlock' ?

**Comment by chinakook** [comment]
How about Swish activation function?

**Comment by anjishnu** [comment]
@kobenaxie I'm not too sure about Blocks vs. HybridBlocks - what would be the difference? If it's just changing the class I inherit from, I can do that right now. 

@chinakook That's a great idea! - Once this is merged we can extend it with other activations.

**Comment by kobenaxie** [comment]
Swish like this ? @chinakook @anjishnu 
```
import mxnet.gluon as gluon
from mxnet import nd

class Swish(gluon.HybridBlock):
    r'''
    Swish Activation function
    https://arxiv.org/pdf/1710.05941.pdf
    Parameters
    ----------
    beta : float
        swish(x) = x * sigmoid(beta*x)
    '''
    def __init__(self, beta=1.0, **kwargs):
        super(Swish, self).__init__(**kwargs)
        self.beta = beta

    def hybrid_forward(self, F, x):
        return x * F.Activation(self.beta * x, act_type='sigmoid', name='fwd')
```
`

**Comment by anjishnu** [comment]
I'll just update the CR to be hybridblocks and include Swish.

**Comment by anjishnu** [comment]
@piiswrong @mseeger please take a look

File: python/mxnet/gluon/contrib/activations.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+''' This file contains definitions of advanced activation functions
+for neural networks'''
+
+import mxnet.gluon as gluon
+
+
+class ELU(gluon.HybridBlock):
+    r'''
+    Exponential Linear Unit (ELU)
+    ... "Fast and Accurate Deep Network Learning by Exponential Linear Units", Clevert et al, 2016
```
Comment (piiswrong) [inline python/mxnet/gluon/contrib/activations.py]: no ... needed

**Comment by piiswrong** [comment]
Could you add unittests?
See tests/python/unittests/test_gluon.py

File: python/mxnet/gluon/contrib/activations.py (hunk 1 above)
Comment (anjishnu) [reply python/mxnet/gluon/contrib/activations.py]: Sorry, what does this comment mean?

**Comment by anjishnu** [comment]
This commit is becoming a bit of a mess. I tried to merge with upstream/master and all those commits have appeared and are polluting this PR. The python2 tests are also failing for some reason which seems completely unrelated to my changes, does anyone know what might be happening here?

**Comment by mseeger** [comment]
This is all a bit of a mess. Why have so many commits? This should be a single commit. Please squash your commits and use rebase instead of merge.

**Comment by piiswrong** [comment]
The original author seems to have disappeared.

Can one of the committers take over?
@yzhliu @szha

**Comment by szha** [comment]
working on it in the above PR.

**Comment by anjishnu** [comment]
Sorry guys, I got a bit busy on some deadlines - need more practice with github PRs and making clean and concise.

**Pull request closed without merging**